### ISSUE TITLE
contract code of huobi_dm  mapping is not correct after settlement

### ISSUE THREAD

**Issue by caizhiwei** [bmoscon/cryptofeed#236]
Pair name of huobi_dm is mapping to its expire date, and this map is read from api when started but it's changed after settlement in every Friday. 
for example:
before 2020-04-10 16:00:
this week  ->  20200410
next week  ->  20200417
after 2020-04-10 16:00:
this week  ->  20200417
next week  ->  20200424
I think this mapping increases unnecessary complexity and should be removed.

**Comment by xiandong79** [comment]
yes

**Comment by bmoscon** [comment]
the same issue would exist with or without the mapping though, right? you'd still need to subscribe to new instruments as time progresses?

**Comment by caizhiwei** [comment]
> the same issue would exist with or without the mapping though, right? you'd still need to subscribe to new instruments as time progresses?

No, subscription to this week will change to next week automatically after settlement. If we remove the mapping, we only need to subscribe once and record data permanently and when we need expired day we can calculate it from data update time and contract type.

**Comment by bmoscon** [comment]
makes sense. I think you can use the contract name directly though right? You dont have to use the mapping do you?

**Comment by caizhiwei** [comment]
> makes sense. I think you can use the contract name directly though right? You dont have to use the mapping do you?
@bmoscon 
It raised the problem when I restarted my program at around 2020-03-22. Before I restarted my program, data from this quarter contract were recorded as 200327, and after I restarted, data from this week were recorded as 200327. Data from different contract were messing up and I couldn't separate them any more because I didn't remember the exact restarted time.
